hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/InProcessHapiTestNode.java

```diff
@@ -74,12 +123,24 @@ public void start() {
     }
 
     @Override
-    public void waitForActive(long seconds) {
+    public void waitForActive(long seconds) throws TimeoutException {
         final var waitUntil = System.currentTimeMillis() + (seconds * 1000);
-        while (th == null || th.hedera == null || !th.hedera.isActive()) {
-            if (System.currentTimeMillis() > waitUntil) {
-                throw new RuntimeException(
-                        "node " + nodeId + ": Waited " + seconds + " seconds, but node did not become active!");
+        while (System.currentTimeMillis() < waitUntil) {
```

Comment:
Might be good to put in a sleep of `(seconds/10)` seconds before this, just to give the node some chance to come up before querying it.
A further sleep on each loop might smooth this out, particularly in the more constrained CI environments.